mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRThumbnailImageImpl.java

```diff
@@ -49,40 +51,90 @@ public MCRThumbnailImageImpl(String implName) {
+                MCRAccessManager.checkPermission(tileInfo.getDerivate(), MCRAccessManager.PERMISSION_VIEW) ||
+                MCRAccessManager.checkPermission(tileInfo.getDerivate(), MCRAccessManager.PERMISSION_READ);
+        }
+        return false;
+    }
+
+    /**
+     * Builds a MyCoRe Object ID from the given IIIF image id parameter.
+     * 
+     * Subclasses may override it.
+     *
+     * @param id the IIIF image id parameter
+     * @return an Optional containing the MyCoRe ID
+     */
+    protected Optional<MCRObjectID> calculateMCRObjectID(String id) {
+        return MCRObjectID.isValid(id) ? Optional.of(MCRObjectID.getInstance(id)) : Optional.empty();
+    }
+
+    private Optional<MCRTileInfo> createTileInfoForMCRObject(MCRObjectID mcrID) {
+        MCRObject mcrObj = MCRMetadataManager.retrieveMCRObject(mcrID);
+        for (MCRMetaEnrichedLinkID derLink : mcrObj.getStructure().getDerivates()) {
+            if (derLink.getClassifications().stream()
+                .map(MCRCategoryID::toString)
+                .anyMatch(derivateTypes::contains)) {
+                final String maindoc = derLink.getMainDoc();
+                if (maindoc != null) {
+                    Optional<MCRTileInfo> tileInfoForFile = createTileInfoForFile(derLink.getXLinkHref(), maindoc);
+                    if (tileInfoForFile.isPresent()) {
+                        return tileInfoForFile;
+                    }
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<MCRTileInfo> createTileInfoForDerivateLink(MCRObjectID mcrID) {
+        Collection<String> linkedImages = MCRLinkTableManager.instance()
+            .getDestinationOf(mcrID, MCRLinkTableManager.ENTRY_TYPE_DERIVATE_LINK);
+        for (String linkedImage : linkedImages) {
+            MCRObjectID deriID = MCRObjectID.getInstance(
+                linkedImage.substring(0, linkedImage.indexOf("/")));
+            MCRDerivate deri = MCRMetadataManager.retrieveMCRDerivate(deriID);
+            if (deri.getDerivate().getClassifications().stream()
+                .map(classi -> classi.getClassId() + ":" + classi.getCategId())
+                .anyMatch(derivateTypes::contains)) {
+                Optional<MCRTileInfo> tileInfoForFile = createTileInfoForFile(deriID.toString(),
+                    linkedImage.substring(linkedImage.indexOf("/") + 1));
+                if (tileInfoForFile.isPresent()) {
+                    return tileInfoForFile;
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<MCRTileInfo> createTileInfoForFile(String derID, String file) {
+        final MCRTileInfo mcrTileInfo = new MCRTileInfo(derID, file, null);
+        final Optional<Path> tileFile = this.tileFileProvider.getTileFile(mcrTileInfo);
+        if (tileFile.isPresent() && Files.exists(tileFile.get())) {
```

Comment:
can be written as return tileFile.filter(Files::exist);

Reply:
I cant, because i need to return the MCRTileInfo and not the Path.

Reply:
The you could use:
```
       final MCRTileInfo tileInfo = new MCRTileInfo(derID, file, null);
       return Optional.of(tileInfo).filter(t-> this.tileFileProvider.getTileFile(t).filter(Files::exists).isPresent());
```